tests/st/runtime/framework_and_models/test_benchmark_st.py

```
@@ -94,5 +96,116 @@ def test_benchmark_register_once_surfaces_timing(test_config, tmp_path):
+        assert stats.device_wall_us[k] == max(v[k] for v in rank_dev.values())
+
+    # Cross-rank host-timeline union window is populated (one per round) and, as a
+    # window spanning all ranks' host spans, is >= the per-round host max.
+    assert len(union) == rounds
+    for k in range(rounds):
+        assert union[k] >= max(v[k] for v in rank_host.values()) - 1e-6
+
+    assert not stats.all_zero_device, "device_wall_us must be > 0 on the default SIMPLER_PROFILING build"
+    assert stats.device_us_min > 0.0
+
+    # Per-card L2 Effective (orch∪sched window): one series per rank, > 0, and
```

Ruff (0.15.20)

[warning] Comment contains ambiguous `∪` (UNION). Did you mean `U` (LATIN CAPITAL LETTER U)?

(RUF003)